### PR TITLE
DOP-2338: Consistent Navigation PoC, 1.0.0 w/ feature flag

### PR DESCRIPTION
**Pull request by casthewiz**
### Stories/Links:

DOP-2338

### Staging Links:
Feature flag on:
https://docs-mongodbcom-staging.corp.mongodb.com/master/docs/cassidy.schaufele/con-nav-poc/

Feature flag off:
https://docs-mongodbcom-staging.corp.mongodb.com/master-flag-off/docs/cassidy.schaufele/con-nav-poc/

### Notes:
- Implements the consistent navigation component with a feature flag to toggle on/off behavior at build time of a docs property.
- Ridealong additional command at `npm build:clean` that runs `npm run clean && npm run build`

### CODE REVIEW

File: src/components/Navbar.js [hunk 1]
```diff
@@ -140,37 +141,51 @@ const Navbar = () => {
   }, [isSearchbarDefaultExpanded]);
 
   return (
-    <ConditionalWrapper
-      condition={theme.bannerContent.enabled}
-      wrapper={(children) => (
+    <>
+      {process.env.FEATURE_FLAG_CONSISTENT_NAVIGATION ? (
```
Comment (casthewiz) [inline src/components/Navbar.js:145]: This env var has to be set to `truthy` within the autobuilder's instances or within `.env` files during development in order for the consistent navigation to render.

If `undefined`, `falsey`, or simply not present in any configuration, our original nav will render.

File: package.json [hunk 2]
```diff
@@ -48,6 +49,9 @@
       "npm run lint:fix"
     ]
   },
+  "publishConfig": {
+    "registry": "https://artifactory.corp.mongodb.com/artifactory/api/npm/npm-local/"
```
Comment (rayangler) [inline package.json:53]: Does this mean that we're deploying snooty to artifactory?

File: src/components/Navbar.js (hunk 1 above)
Comment (rayangler) [reply src/components/Navbar.js:145]: Great idea. Is there any more work that needs to be done on our end for the consistent nav? Or are we just waiting for the go signal to remove the flag?

File: src/components/Navbar.js [hunk 3]
```diff
@@ -140,37 +141,51 @@ const Navbar = () => {
...
         <div
           css={css`
             position: fixed;
             top: 0;
+            width: 100%;
+            z-index: 2;
           `}
         >
-          <Banner {...theme.bannerContent} />
```
Comment (rayangler) [inline src/components/Navbar.js:152]: Lint warnings on where we assign the banner! We probably won't have to set up another banner any time soon so I'm okay with removing its assignment as well

**Review by rayangler — COMMENTED** [review]
Just a couple of questions for clarification!

File: package.json (hunk 2 above)
Comment (casthewiz) [reply package.json:53]: Nope - I was instructed to add it when we first started integrating with the nav package, but I'm actually curious if it's needed. I'll look at dropping it. 🧐

Comment (casthewiz) [reply package.json]: I was able to safely drop it 🤷

**Comment by rayangler** [comment]
I noticed that the "Try Free" anchor loads its green background with a delay (not sure why), and hovering over the link makes the text blue and underlines it. 

I'm convinced that the latter may be happening due to our global css file. Is there a way we can override the styling for that anchor tag specifically?

**Comment by casthewiz** [comment]
> I noticed that the "Try Free" anchor loads its green background with a delay (not sure why), and hovering over the link makes the text blue and underlines it.
> 
> I'm convinced that the latter may be happening due to our global css file. Is there a way we can override the styling for that anchor tag specifically?

So the green background delay seems intentional - there's a transition effect on that part of the component. As far as overriding the styling, I've tried a few ways of resolving but none were really satisfactory. Either we risk leaking the overriding styles to other `a` tags in the shared component, or we risk having a "gotcha" style that overwrites any styles they might add in future versions.

I'm going to flag that `:hover` styles should be explicitly set on that button for future versions, to avoid this situation 👍

**Review by rayangler — APPROVED** [review]
LGTM! Thanks for looking into the styling

**Review by mango-db — APPROVED** [review]
LGTM!

Looks good on:
- [x] Safari
- [x] Chrome
- [x] Firefox

**Pull request merged**